src/pages/signup/Signup.jsx

```diff
@@ -0,0 +1,115 @@
+import React from "react";
+import { Link } from "react-router-dom";
+import {
+  PasswordNotShowIcon,
+  PasswordShowIcon,
+} from "../../assests/icons/icons";
+import { useAuthContext } from "../../context";
+
+import "./Signup.css";
+
+export function Signup() {
+  const { signupHandler, state, dispatch } = useAuthContext();
+
+  return (
+    <div className="signup-page">
+      <form
+        className="form form-signup"
+        onSubmit={(e) => {
+          signupHandler(e);
+        }}
+      >
+        <h2 className="title-form">Signup</h2>
+        <div className="parent-div">
+          <input
+            type="text"
+            placeholder="first name"
+            className="form-input flex-1"
+            name="firstName"
+            onChange={(e) => dispatch({ type: "ADD_FIELD", payload: e.target })}
+            required
+          />
+          <input
+            type="text"
+            placeholder="last name"
+            className="form-input flex-1"
+            name="lastName"
+            onChange={(e) => dispatch({ type: "ADD_FIELD", payload: e.target })}
+            required
+          />
+        </div>
+        <input
+          type="text"
+          placeholder="email"
+          className="form-input"
+          name="email"
+          onChange={(e) => dispatch({ type: "ADD_FIELD", payload: e.target })}
+          required
+        />
+        {state.emailErrState && (
+          <small className="form-error">invalid mail</small>
+        )}
+        <div className="parent-div">
+          <input
+            type={state.passwordType}
+            placeholder="password"
+            className="form-input flex-1"
+            name="password"
+            onChange={(e) => dispatch({ type: "ADD_FIELD", payload: e.target })}
+            required
+          />
+          <span
```

Comment:
Instead of `span` here, try using `button`. It will make the code more semantic and will increase its accessibility.